api/api.go

```diff
@@ -248,6 +248,28 @@ func (c *Client) GetTeamName(teamUUID uuid.UUID) (string, error) {
 	return decodedJSON.Name, nil
 }
 
+// CreateRequestToJoinTeam posts a request to join the team identified by the UUID with the
+// given fingerprint and email
+func (c *Client) CreateRequestToJoinTeam(
+	teamUUID uuid.UUID, fpr fingerprint.Fingerprint, email string) (err error) {
+
+	path := fmt.Sprintf("team/%s/requests-to-join", teamUUID)
+	requestToJoinTeamRequest := v1structs.RequestToJoinTeamRequest{TeamEmail: email}
+
+	request, err := c.newRequest("POST", path, requestToJoinTeamRequest)
+	if err != nil {
+		return err
+	}
+	request.Header.Add("authorization", authorization(fpr))
+
+	_, err = c.do(request, nil)
```

Comment:
Currently no handling of the special `409` cases, is that OK?

Reply:
I've added that now.